app/domain/operations/fdsh/payload_eligibility/check_application_eligibility_rules.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+module Operations
+  module Fdsh
+    module PayloadEligibility
+      # Inherits from CheckEligibilityRules
+      class CheckApplicationEligibilityRules < CheckBaseEligibilityRules
+
+        private
+
+        def validate(payload_entity, request_type)
+          return Failure("Invalid Application Object #{payload_entity}") unless payload_entity.is_a?(AcaEntities::MagiMedicaid::Application)
+          super(request_type)
+        end
+
+        def validate_ssn(payload)
+          result = payload.applicants.map do |applicant|
+            encrypted_ssn = applicant.identifying_information.encrypted_ssn
+            return Failure("No SSN for applicant") if encrypted_ssn.nil? || encrypted_ssn.empty?
+
+            result = Operations::Fdsh::EncryptedSsnValidator.new.call(encrypted_ssn)
+            return Failure("Invalid SSN") unless result.success?
```

Comment:
EncryptedSsnValidator returns respective error message. This line can be removed.

```suggestion

```

Reply:
also remove 'result = '

Reply:
So I ended up refactoring this to 

`result = Operations::Fdsh::EncryptedSsnValidator.new.call(encrypted_ssn)`
`return result unless result.success?`

because I believe all applicants on an application must have valid SSNs before an FA application can be submitted, correct? Just doing `Operations::Fdsh::EncryptedSsnValidator.new.call(encrypted_ssn)` would return nil -> I think an explicit return here might work better to break out of the iterator on the first error.

Let me know what you think.